contracts/message-bus/Proof.sol

```diff
@@ -0,0 +1,266 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../lib/CircularBufferUint.sol";
+import "./StateRootI.sol";
+import "../lib/RLP.sol";
+import "../lib/MerklePatriciaProof.sol";
+import "../lib/BytesLib.sol";
+
+contract Proof is CircularBufferUint{
+
+    /* Variables */
+
+    /** Contract address for which the storage is to be proved. */
+    address public storageAccount;
+
+    /** State root provider address. */
+    StateRootI public stateRootProvider;
+
+    /** Encoded account path. */
+    bytes public encodedAccountPath;
+
+    /** Maps blockHeight to storageRoot. */
+    mapping(uint256 => bytes32) public storageRoots;
+
+
+    /* Internal Functions */
+
+    /**
+     * @notice Setup the proof contract. This can be called only once.
+     *
+     * @dev Function requires:
+     *          - proof contract must not be already initialized
+     *          - storageAccount must not be zero
+     *          - stateRootProvider must not be zero
+     *
+     * @param _storageAccount Storage account that will be proved.
+     * @param _stateRootProvider State root provider contract address.
+     * @param _maxStorageRootItems Defines how many storage roots should be
+     *                             stored in circular buffer.
+     */
+    function initialize(
```

Comment:
Here name is intentionally deviated from other contracts i.e. `setup**`?

Reply:
Changes to setup